decidim-admin/app/commands/decidim/admin/participatory_space/unpublish.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Admin
+    module ParticipatorySpace
+      class Unpublish < Decidim::Command
+        # Public: Initializes the command.
+        #
+        # participatory_space - A participatory_space that will be published
+        # user - the user performing the action
+        def initialize(participatory_space, user)
+          @participatory_space = participatory_space
+          @user = user
+        end
+
+        # Executes the command. Broadcasts these events:
+        #
+        # - :ok when everything is valid.
+        # - :invalid if the data was not valid and we could not proceed.
+        #
+        # Returns nothing.
+        def call
+          return broadcast(:invalid) if participatory_space.nil? || !participatory_space.published?
+
+          Decidim.traceability.perform_action!(:unpublish, participatory_space, user, **default_options) do
+            participatory_space.unpublish!
+          end
+
+          broadcast(:ok, participatory_space)
+        end
+
+        private
+
+        attr_reader :participatory_space, :user
+
+        def default_options = {}
```

Comment:
Related to my comment in `Decidim::Votings::Admin::UnpublishVoting`, does it make any sense to have `{ visibility: "all" }` in the `Decidim::Admin::ParticipatorySpace::Publish` command but not in the `Decidim::Admin::ParticipatorySpace::Unpublish`?

I know that you're respecting current behavior (i.e. how it works now in `UnpublishAssembly`, `UnpublishConference` and `UnpublishParticipatoryProcess`) but maybe it's time to make it consistent. Any thoughts?

Reply:
Let's make it consistent then :)